python/liblrs_python.pyi

```diff
@@ -0,0 +1,148 @@
+# This file is automatically generated by pyo3_stub_gen
+# ruff: noqa: E501, F401
+
+import typing
+
+class Anchor:
+    r"""
+    An `Anchor` is a reference point for a given [`Curve`].
+    It can be a milestone, a bridge…
+    """
+    name: str
+    position: typing.Optional[Point]
+    curve_position: float
+    scale_position: float
+
+class AnchorOnLrm:
+    r"""
+    The linear position of an anchor doesn’t always match the measured distance
+    For example if a road was transformed into a bypass, resulting in a longer road,
+    but measurements are kept the same
+    The start of the curve might also be different from the `0` of the LRM
+    """
+    def __new__(cls,anchor_index:int, distance_along_lrm:float): ...
+    ...
+
+class Builder:
+    def __new__(cls,): ...
+    def add_node(self, id:str, coord:Point, properties:typing.Mapping[str, str]) -> int:
+        ...
+
+    def add_anchor(self, id,coord,properties,name = ...) -> int:
+        ...
```

Comment:
Why did `add_node` get some typing for `coord`, and not `add_anchor` ?

Reply:
Oh no, another rabbithole :sweat_smile: 

Because I declared a type `#[pyo3(signature = (id, coord, properties, name=None))]` to have default values for name. I’ll dig a bit into that

Reply:
So… somebody opened an issue about that… yesterday : https://github.com/Jij-Inc/pyo3-stub-gen/issues/82

It will probably take some time until full generation, but for now it’s in the stub